src/shapes/image.class.ts

```diff
@@ -1,869 +1,797 @@
-       * and refreshed at the next render
-       * @type Array
-       */
-      cacheProperties: FabricObject.prototype.cacheProperties.concat(
-        'cropX',
-        'cropY'
-      ),
+  /**
+   * private
+   * contains last value of scaleX to detect
+   * if the Image got resized after the last Render
+   * @type Number
+   */
+  protected _lastScaleX = 1;
 
-      /**
-       * key used to retrieve the texture representing this image
-       * @since 2.0.0
-       * @type String
-       * @default
-       */
-      cacheKey: '',
-
-      /**
-       * Image crop in pixels from original image size.
-       * @since 2.0.0
-       * @type Number
-       * @default
-       */
-      cropX: 0,
-
-      /**
-       * Image crop in pixels from original image size.
-       * @since 2.0.0
-       * @type Number
-       * @default
-       */
-      cropY: 0,
-
-      /**
-       * Indicates whether this canvas will use image smoothing when painting this image.
-       * Also influence if the cacheCanvas for this image uses imageSmoothing
-       * @since 4.0.0-beta.11
-       * @type Boolean
-       * @default
-       */
-      imageSmoothing: true,
-
-      /**
-       * Constructor
-       * Image can be initialized with any canvas drawable or a string.
-       * The string should be a url and will be loaded as an image.
-       * Canvas and Image element work out of the box, while videos require extra code to work.
-       * Please check video element events for seeking.
-       * @param {HTMLImageElement | HTMLCanvasElement | HTMLVideoElement | String} element Image element
-       * @param {Object} [options] Options object
-       * @return {fabric.Image} thisArg
-       */
-      initialize: function (element, options) {
-        options || (options = {});
-        this.filters = [];
-        this.cacheKey = 'texture' + FabricObject.__uid++;
-        this.callSuper('initialize', options);
-        this._initElement(element, options);
-      },
-
-      /**
-       * Returns image element which this instance if based on
-       * @return {HTMLImageElement} Image element
-       */
-      getElement: function () {
-        return this._element || {};
-      },
-
-      /**
-       * Sets image element for this instance to a specified one.
-       * If filters defined they are applied to new image.
-       * You might need to call `canvas.renderAll` and `object.setCoords` after replacing, to render new image and update controls area.
-       * @param {HTMLImageElement} element
-       * @param {Object} [options] Options object
-       * @return {fabric.Image} thisArg
-       * @chainable
-       */
-      setElement: function (element, options) {
-        this.removeTexture(this.cacheKey);
-        this.removeTexture(this.cacheKey + '_filtered');
-        this._element = element;
-        this._originalElement = element;
-        this._initConfig(options);
```

Comment:
mmm this is a weird change.
If i do setElement today, i can pass to options random things and it will be applied by setOptions.
Now we locked to width/height.
To be fair we should leave the setOptions call in.
Should we move set(options) inside here? so we maintain the old logic, or we should remove size options.

Reply:
I think simplification is better.
setElement should set the element.
We can comment that setElement will reset width/height but not cropX and cropY ( that is weird by itself )

Reply:
I did cause of simplification, too many side effects are confusing. The dev can call set whenever. `setElement` should focus on its task. It makes sense that setting dimensions is part of its scope